nautilus_core/common/src/throttler/mod.rs

```diff
@@ -42,11 +43,11 @@ impl RateLimit {
 /// Throttler takes messages of type T and callback of type F for dropping
 /// or processing messages.
 #[derive(Clone)]
-pub struct Throttler<T, F> {
-    inner: Rc<RefCell<InnerThrottler<T, F>>>,
+pub struct Throttler<T, S, D> {
+    inner: Rc<RefCell<InnerThrottler<T, S, D>>>,
 }
 
-impl<T, F> Debug for Throttler<T, F>
+impl<T, S, D> Debug for Throttler<T, S, D>
```

Comment:
Why do we need different parameters for Send and Drop when they are the same traits `Fn(T)`? Have you considered removing it?

Reply:
Previously, I was registering two different functions with the same parameters, which was causing an error. The compiler requires both functions to be identical, so I made this change. However, since we're now using closures as callbacks, this is no longer needed, so I have removed it.